Commit ID: 19560604f06dd09031ec4625f5c6678547954f94
Change ID: pwyrsrswtpkmkvnmvxlpvqnyrlvkvslu
Author   : Skyler Grey <sky@a.starrysky.fyi> (2025-07-02 08:22:07)
Committer: Skyler Grey <sky@a.starrysky.fyi> (2025-07-02 08:22:09)

    fix: correct SSL authentication

    We need to auth as a client not as a server here - the arguments are
    rather confusingly named

    Unfortunately this also throws into question the testing/etc. of the SIP
    library: this code could never have worked

    See-Also: https://stackoverflow.com/a/78863325/12293760

diff --git a/PySIP/sip_core.py b/PySIP/sip_core.py
index eb3c26bf5c..e781da717c 100644
--- a/PySIP/sip_core.py
+++ b/PySIP/sip_core.py
@@ -204,7 +204,7 @@
                 self.is_running.set()
 
             elif self.connection_type == ConnectionType.TLS:
-                ssl_context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+                ssl_context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH)
                 if SAVE_TLS_KEYLOG:
                     ssl_context.keylog_filename = "tls_keylog.log"
                 self.reader, self.writer = await asyncio.open_connection(
